### PR TITLE
Rename `devices` to `detect` command line option

### PR DESCRIPTION
**Pull request by jainvikas8**
### Description

To ensure good usability with the new Mbed tools the command line
options need to harmonize with the old tools.

Therefore change `devices` to `detect` by keeping the existing
functionality.

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

**Review by rwalton-arm — APPROVED** [review]

**Comment by Patater** [comment]
The test failures indicate that there are some tests that try to use the CLI command by the old name still.

**Comment by Patater** [comment]
The kvstore failure in Travis CI is due to https://github.com/ARMmbed/mbed-os-example-kvstore/pull/60 being merged, but not yet https://github.com/ARMmbed/mbed-os/pull/13908 and https://github.com/ARMmbed/mbed-os-example-kvstore/pull/61

**Comment by 0xc0170** [comment]
> The kvstore failure in Travis CI is due to ARMmbed/mbed-os-example-kvstore#60 being merged, but not yet ARMmbed/mbed-os#13908 and ARMmbed/mbed-os-example-kvstore#61

I'll test kvstore PR but might revert the example one as we did not manage to release today so we would be blocked until monday (master unblocked). I'll talk to core team.

**Pull request merged**
